guildwars2/key.py

```diff
@@ -63,35 +63,20 @@ class KeyMixin:
             "permissions": token["permissions"]
         }
         #at this point we know the key is valid
-        try:
-            keys = doc["keys"]
-            #check if key is already in keys list
-            if newkeydoc not in keys:
-                if len(keys) < 8:
-                    keys.append(newkeydoc)
-                else:
-                    await ctx.send("You've reached the maximum limit of 8 API keys," 
-                        "please remove one before adding another. {0}".format(output))
-                    return
-            else:
-                await ctx.send("You have already added this key before. {0}".format(output))
-                return
-        except:
-            #keys list doesn't exist - this is either the first key the user has added or 
-            #they existed before this functionality
-            try:
-                currentkey = await self.fetch_key(user)
-                #will raise an exception if the user has no key
-                #if not we should append both keys to the keys list
-                keys = []
-                keys.append(currentkey)
+        keys = doc.get("keys", [])
+        key = doc.get("key", {})
+        if keys == [] and key != {}:
```

Comment:
Nitpicky, but the second argument in dict.get() is optional and defaults to None, then you can use ```py
if not keys and key```, later too